### ISSUE TITLE
S3 entry results in panic with sof kernel

### ISSUE THREAD

**Issue by sathyap-chrome** [thesofproject/linux#832]
On CML, when SOF driver was loaded and DSP fw was not loaded (different issue)
The backtrace is as below. let me know if there is a quick fix i can try to quickly unblock other tests.

get the following crash when I attempt suspend_stress_test -c1

```
[   73.850788] Suspending console(s) (use no_console_suspend to debug)

Entering kdb (current=0xffff88837df8e640, pid 3724) on processor 2 Oops: (null)
due to oops @ 0xffffffff813984e6
CPU: 2 PID: 3724 Comm: cat Tainted: G    B             4.19.34 #9
Hardware name: Google Hatch/Hatch, BIOS Google_Hatch.12062.0.2019_04_11_1841 04/12/2019
RIP: 0010:do_raw_spin_lock+0xc4/0x1e9
Code: 45 00 65 8b 0c 25 20 81 01 00 39 c8 0f 84 ad 00 00 00 48 89 5d c8 4c 89 e7 be 04 00 00 00 e8 fd 2f 35 00 b9 01 00 00 00 31 c0 <f0> 41 0f b1 0c 24 85 c0 0f 85 9b 00 00 00 65 8b 1c 25 20 81 01 00
RSP: 0018:ffff8883a6d4ef58 EFLAGS: 00010046
RAX: 0000000000000000 RBX: ffffffff816edb5c RCX: 0000000000000001
RDX: 0000000000000001 RSI: 0000000000000004 RDI: ffffffff816edb4c
RBP: ffff8883a6d4ef90 R08: dffffc0000000000 R09: fffffbfff02ddb6a
R10: 0000000000000001 R11: fffffbfff02ddb6a R12: ffffffff816edb4c
R13: ffffffff816edb54 R14: 1ffffffff02ddb6b R15: 1ffffffff02ddb6a
FS:  00007a1069bb0840(0000) GS:ffff88840ee80000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: ffffffff816edb4c CR3: 00000003bb908002 CR4: 00000000003606e0
Call Trace:
 ? quarantine_reduce+0x18b/0x190
 ? quarantine_reduce+0x17b/0x190
 sof_ipc_tx_message+0xc6/0xd3a [snd_sof]
 ? quarantine_reduce+0x173/0x190
 ? rpm_resume+0xaf5/0x10a4
 ? rpm_callback+0x1ed/0x1ed
 ? snd_sof_fw_unload+0xb/0xb [snd_sof]
 ? dev_printk_emit+0xd1/0x125
 ? rpm_resume+0xb11/0x10a4
 sof_suspend+0x2c4/0x498 [snd_sof]
 ? snd_sof_runtime_suspend+0x11/0x11 [snd_sof]
 ? _raw_spin_unlock+0xde/0xde
 ? klist_next+0x1e9/0x221
 ? do_raw_spin_unlock+0x4f/0x24e
 ? snd_sof_resume+0x11/0x11 [snd_sof]
 pci_pm_suspend+0x2cb/0x59f
 ? pci_pm_complete+0x1ac/0x1ac
 dpm_run_callback+0x216/0x4be
 ? pci_pm_complete+0x1ac/0x1ac
 ? dpm_subsys_suspend_noirq_cb+0x2cc/0x2cc
 __device_suspend+0xa0f/0xf4e
 ? deactivate_slab+0x2b6/0x2e5
 ? async_suspend+0xe4/0xe4
 ? __might_sleep+0xfc/0xfc
 ? wq_worker_sleeping+0x235/0x235
 ? mutex_lock+0xe0/0x18c
 ? do_wait_for_common+0x5c2/0x5c2
 ? kobject_get+0x88/0x88
 dpm_suspend+0x347/0x92e
 ? dpm_suspend_end+0x7e/0x7e
 ? console_lock+0x44/0x44
 suspend_devices_and_enter+0x1bb/0x15f4
 ? arch_suspend_enable_irqs+0xc/0xc
 ? wake_up_klogd+0x9a/0x114
 ? console_unlock+0xbc9/0xbc9
 ? cont_add+0xca/0xeb
 ? vprintk_store+0x1e6/0x28b
 ? printk+0xd9/0x12d
 ? pm_suspend+0xa00/0x11ca
 ? log_buf_vmcoreinfo_setup+0x131/0x131
 ? do_raw_spin_unlock+0x4f/0x24e
 ? __pm_pr_dbg+0x186/0x190
 ? pm_suspend+0xa00/0x11ca
 ? pm_notifier_call_chain+0x39/0x39
 ? bsp_pm_callback+0x2a/0x48
 pm_suspend+0xb65/0x11ca
 ? suspend_test+0x9d/0x9d
 ? deactivate_slab+0x2b6/0x2e5
 ? mutex_lock+0xe0/0x18c
 ? kernfs_get_active+0xf0/0x257
 ? do_wait_for_common+0x5c2/0x5c2
 ? kernfs_get+0x66/0x66
 state_store+0xd8/0x147
 ? sysfs_kf_read+0x182/0x182
 kernfs_fop_write+0x255/0x341
 __vfs_write+0x108/0x608
 ? vfs_read+0x2e9/0x2e9
 ? file_has_perm+0x1e4/0x2b5
 ? __sb_start_write+0x9b/0x283
 ? __sb_end_write+0x17a/0x17a
 ? security_file_permission+0x54/0x155
 vfs_write+0x1aa/0x434
 ksys_write+0xfe/0x1be
 ? __ia32_sys_read+0x82/0x82
 ? vmalloc_sync_all+0x23/0x23
 do_syscall_64+0xc8/0x11b
 entry_SYSCALL_64_after_hwframe+0x44/0xa9
RIP: 0033:0x7a1069649351
Code: c3 0f 1f 84 00 00 00 00 00 48 8b 05 f1 7b 20 00 c3 0f 1f 84 00 00 00 00 00 8b 05 5a be 20 00 85 c0 75 2e b8 01 00 00 00 0f 05 <48> 3d 00 f0 ff ff 77 07 c3 66 0f 1f 44 00 00 48 8b 15 f9 79 20 00
RSP: 002b:00007ffd6b3f1ae8 EFLAGS: 00000246 ORIG_RAX: 0000000000000001
RAX: ffffffffffffffda RBX: 00007a1069b8f000 RCX: 00007a1069649351
RDX: 0000000000000006 RSI: 00007a1069b8f000 RDI: 0000000000000001
RBP: 00007ffd6b3f1b10 R08: 00000000ffffffff R09: 0000000000000000
R10: 0000000000000022 R11: 0000000000000246 R12: 0000000000000006
R13: 0000000000020000 R14: 00007a1069b8f000 R15: 0000000000000001
more>
```
I will see if the same thing happens with S3

SEEN ON : CML Chrome platform
SOF REL : v0.2

**Comment by xiulipan** [comment]
@sathyap-chrome 
Duplicate of #505 #494
This is fixed with https://github.com/thesofproject/linux/pull/619

**Comment by sathyap-chrome** [comment]
Thanks.

**Comment by wenqingfu** [comment]
resolved? @sathyap-chrome

**Comment by wenqingfu** [comment]
> @sathyap-chrome
> Duplicate of #505 #494
> This is fixed with #619

close the issue
